compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/MembersInjectorGenerator.kt

```diff
@@ -91,7 +93,7 @@ internal class MembersInjectorGenerator : PrivateCodeGenerator() {
         .joinToString()
     }
 
-    val memberInjectorClass = ClassName(packageName, className)
+    val memberInjectorClass = classId.asClassName()
```

Comment:
What's the difference between this and `className` (classId.relativeClassName.asString())? If this one is the fully qualified name, it might help with clarity to rename `className` more explicitly. I found the vals all blurring together while reading 😅

Reply:
A class has a package and a class name (sometimes multiple class names for inner classes. FqName can't differentiate between the package and class name, because `FqName` is basically a string, e.g. `com.squareup.Outer.Inner`. We usually consider the first element starting with an uppercase letter as the first class name, so we'd treat `com.squareup` as package and `Outer.Inner` as relative class name. That approach has problems, if you use uppercase letters in a package name and unfortunately that happens (what's wrong with those people?!).

`ClassId` (Kotlin compiler type) basically holds two FqNames: one for the package and one for the class names. `relativeClassName` would be `Outer.Inner` in this sample. I prefer `ClassId` over `FqName` any day.

`ClassName` is a type from KotlinPoet and very similar to `ClassId`. It needs to know the package and class name. Converting a `ClassId` to `ClassName` is simple, but converting a `FqName` to `ClassName` could go wrong. 

The old `generateClassName` method returned basically the relative class name only and not the package. Since I changed it to a `ClassId` we now have richer information, but therefore also must call `relativeClassName` to restore the old behavior. 

I hope that makes sense 😅

Reply:
Yes that makes things much clearer, thanks for the thorough explanation!